nats/errors.py

```diff
@@ -65,8 +69,15 @@ def __str__(self) -> str:
 
 
 class SlowConsumerError(Error):
-    def __init__(self, subject=None, sid=None, sub=None) -> None:
+    def __init__(
+        self,
+        subject: Optional[str] = None,
+        reply: Optional[str] = None,
+        sid: Optional[int] = None,
+        sub: Subscription = None
```

Comment:
Well, in fact I've been wondering about those annotations.

We use `Optional` solely because default value is `None` in `__init__` method, but `SlowConsumerError` will always have a `sub` attribute defined and not None.

The only field which should be `Optional` IMO is `reply`. I think that other fields should not have a default value and not be `Optional`.

What are your thoughts on this ?

Reply:
Agree with @charbonnierg here, `reply` would be the only optional one and the other ones should not have a default value of None which makes them optional...

Reply:
Indeed, I remember that I found no use case of default values either before.

It should be great to change the function signature, explicitly expecting non None values! (other than reply)

Reply:
In fact, event `reply` should not be None after giving it some thoughts, because it's taken from ` msg.reply` which is never `None` but can be equal to an epty string (`""`).

So it should be simply `str` and not `Optional[str]`. Users will still be able to check if a reply should be sent using `if error.reply: ...`

Reply:
I updated the code according to my last comment

Reply:
Thanks!